### PR TITLE
feat: add support to restart identity on truncate

### PR DESCRIPTION
**Pull request by RafaelGSS**
ref #41 

Not sure about the implementation itself, I never played with Nearley before.

### CODE REVIEW

**Comment by RafaelGSS** [comment]
Actually, It translates `truncate table example restart identity` to `truncate table example` that is not expected. Any thoughts?

File: src/syntax/delete.ne [hunk 1]
```diff
@@ -28,7 +28,7 @@ delete_delete -> (kw_delete %kw_from)
                         });
                     } %}
 
-delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] {% x => track(x, {
+delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] (kw_restart kw_identity):? {% x => track(x, {
                             type: 'truncate table',
                             tables: x[1],
-                        }) %}
```
Comment (oguimbal) [inline src/syntax/delete.ne:34]: Technically, that parses, but the fact that `restart identity` is present should influence the generated AST.

In another hand, when it is NOT present, the generated AST should be the same as previously.

Moreover, if you implement `restart identity`, you should also implement `continue identity` ... see [the spec](https://www.postgresql.org/docs/current/sql-truncate.html).

Meaning something like: 

```typescript
delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] ((kw_restart | kw_continue) kw_identity):? {% x => track(x, {
                            type: 'truncate table',
                            tables: x[1],
                            ...x[2] && { identity: toStr(x[2][0]) },
                        })
```

and modify the AST definition:

```typescript
export interface TruncateStatement {
    //  [.....]
    identity?: 'restart' | 'continue';
}
```

**Comment by oguimbal** [comment]
Thanks :)

**Comment by oguimbal** [comment]
Released as `pg-mem@8.0.0`  along with some non trivial breaking changes (sorry for that if it impacts you, I'll create a release notes)

**Comment by oguimbal** [comment]
see [here](https://github.com/oguimbal/pgsql-ast-parser/discussions/43)

**Comment by RafaelGSS** [comment]
@oguimbal any plans to use it in pg-mem?

**Pull request merged**